lms/djangoapps/completion/tests/test_models.py

```diff
@@ -142,3 +142,30 @@ def test_cannot_call_submit_completion(self):
                 completion=0.9,
             )
         self.assertEqual(models.BlockCompletion.objects.count(), 1)
+
+
+class SubmitBatchCompletionTestCase(TestCase):
```

Comment:
Please add tests for all your validation error cases.  To get 100% coverage, you need to test each branch and possibility.